pyodide_build/recipe/cleanup.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from pathlib import Path
+
+from pyodide_build.logger import logger
+from pyodide_build.recipe import loader
+
+
+def resolve_targets(
+    recipe_dir: Path,
+    names_or_tags: Iterable[str] | None,
+    *,
+    include_always_tag: bool = False,
+) -> list[str]:
+    """
+    Resolve package names from names/tags using the recipe loader.
+
+    If names_or_tags is None, selects all packages ("*").
+    By default, packages with the "always" tag are not implicitly included.
+    """
+    if names_or_tags is None:
+        names_or_tags = ["*"]
+
+    recipes = loader.load_recipes(
+        recipe_dir, names_or_tags, load_always_tag=include_always_tag
+    )
+    return list(recipes.keys())
+
+
+def remove_package_build(recipe_dir: Path, build_dir_base: Path, package: str) -> bool:
```

Comment:
All the `remove_...` functions are doing the similar thing with duplicate codes, so it would be nice to refactor the functions to be more general.

I think we can have two functions that:

1. Removes any file or directory given as a parameter
2. Locate a set of files to remove

Then, the function 1 would have a more simple signature, and can be used in a more general purposes.

If we have more files to delete in the future, we can update function 2 to include those files.